### ISSUE TITLE
clj-kondo marks args in `definterface` as unused

### ISSUE THREAD

**Issue by dpsutton** [clj-kondo/clj-kondo#1871]
**version**

clj-kondo v2022.08.03

**platform**

native

**problem**

Clj-kondo marks args in `definterface` as unused

**repro**

```shell
❯ echo "(definterface Foo (foo [x]) (bar [x]))" | clj-kondo --lint -
<stdin>:1:35: warning: unused binding x
linting took 12ms, errors: 0, warnings: 1
```

The example from `(doc definterface)`:
<img width="374" alt="image" src="https://user-images.githubusercontent.com/6377293/200592194-df82642a-1adb-41ac-bf94-0f1c463259a6.png">

The two args in the second method are marked as unused (ignore the unresolved type names). Interestingly, the args in the first method are not marked as unused.

**expected behavior**

There should be no unused args warning for definterface.
